### ISSUE TITLE
AppVeyor CI timing out on windows

### ISSUE THREAD

**Issue by inetic** [maidsafe/crust#393]
Seems to have started with [this PR](https://github.com/maidsafe/crust/pull/376) (the CI machined did not report it as failure at the time).

**Comment by inetic** [comment]
Update: Fraser seems to be able to reproduce this if he enables two network interfaces in his windows PC:

>  I was just running the exact commands which the CI machines run to see if I could replicate the hanging on Windows. I can make it hang every time if I enable 2 network cards on my machine.  If I only enable 1 card, the tests always exit cleanly.

**Comment by inetic** [comment]
It also only happens if all the tests are executed (`cargo test`), if one executes the tests one by one, they all succeed.

```
cargo test --release -- --nocapture beacon::test::test_avoid_beacon
cargo test --release -- --nocapture beacon::test::test_beacon
cargo test --release -- --nocapture bootstrap_handler::test::duplicates
cargo test --release -- --nocapture bootstrap_handler::test::max_contacts
cargo test --release -- --nocapture bootstrap_handler::test::prune
cargo test --release -- --nocapture bootstrap_handler::test::serialise_and_parse
cargo test --release -- --nocapture config_handler::test::parse_sample_config_file
cargo test --release -- --nocapture config_handler::test::read_config_file_test
cargo test --release -- --nocapture file_handler::test::read_write_file_test
cargo test --release -- --nocapture getifaddrs::test::test_filter_loopback
cargo test --release -- --nocapture getifaddrs::test::test_getifaddrs
cargo test --release -- --nocapture hole_punching::tests::test_get_mapped_socket_from_self
cargo test --release -- --nocapture hole_punching::tests::test_udp_hole_punching_0_time_out
cargo test --release -- --nocapture hole_punching::tests::test_udp_hole_punching_1_terminates_no_secret
cargo test --release -- --nocapture hole_punching::tests::test_udp_hole_punching_2_terminates_with_secret
cargo test --release -- --nocapture map_external_port::test::upnp ... ignored
cargo test --release -- --nocapture sequence_number::test::sequence_number
cargo test --release -- --nocapture service::test::bootstrap
cargo test --release -- --nocapture service::test::connection_manager
cargo test --release -- --nocapture service::test::connection_manager_start
cargo test --release -- --nocapture service::test::network
cargo test --release -- --nocapture service::test::reaccept
cargo test --release -- --nocapture service::test::rendezvous_connection
cargo test --release -- --nocapture state::test::bootstrap_off_list
cargo test --release -- --nocapture tcp_connections::test::send_messages_fast
cargo test --release -- --nocapture tcp_connections::test::test_multiple_nodes_small_stream
cargo test --release -- --nocapture tcp_connections::test::test_small_stream
cargo test --release -- --nocapture test::check_rust_unit_testing_is_not_parallel
cargo test --release -- --nocapture transport::test::test_cbor
cargo test --release -- --nocapture transport::test::test_ord
cargo test --release -- --nocapture util::test::test_heuristic_geo_cmp
cargo test --release -- --nocapture utp_connections::test::cannot_establish_connection
cargo test --release -- --nocapture utp_connections::test::establishing_connection
cargo test --release -- --nocapture utp_connections::test::send_receive_data
```

**Comment by vinipsmaker** [comment]
Fixed by #400
